core/execution/common/market_activity_tracker.go

```diff
@@ -582,28 +585,34 @@ func (mat *MarketActivityTracker) CalculateMetricForTeams(ds *vega.DispatchStrat
 	stakingRequirement, _ := num.UintFromString(ds.StakingRequirement, 10)
 	notionalRequirement, _ := num.UintFromString(ds.NotionalTimeWeightedAveragePositionRequirement, 10)
 	topNDecimal := num.MustDecimalFromString(ds.NTopPerformers)
-	return mat.calculateMetricForTeams(ds.AssetForMetric, teamMembers, ds.Markets, ds.Metric, stakingRequirement, notionalRequirement, int(ds.WindowLength), topNDecimal)
+
+	p, _ := lproto.Marshal(ds)
+	gameID := hex.EncodeToString(crypto.Hash(p))
+
+	return mat.calculateMetricForTeams(ds.AssetForMetric, teamMembers, ds.Markets, ds.Metric, stakingRequirement, notionalRequirement, int(ds.WindowLength), topNDecimal, gameID)
 }
 
-func (mat *MarketActivityTracker) isEligibleForReward(asset, party string, markets []string, minStakingBalanceRequired *num.Uint, notionalTimeWeightedAveragePositionRequired *num.Uint) bool {
+func (mat *MarketActivityTracker) isEligibleForReward(asset, party string, markets []string, minStakingBalanceRequired *num.Uint, notionalTimeWeightedAveragePositionRequired *num.Uint, gameID string) bool {
 	if !minStakingBalanceRequired.IsZero() {
 		balance, err := mat.balanceChecker.GetAvailableBalance(party)
 		if err != nil || balance.LT(minStakingBalanceRequired) {
 			return false
 		}
 	}
 	if !notionalTimeWeightedAveragePositionRequired.IsZero() {
-		if mat.getTWNotionalPosition(asset, party, markets).LT(notionalTimeWeightedAveragePositionRequired) {
+		notional := mat.getTWNotionalPosition(asset, party, markets)
```

Comment:
what about the case it is zero?

Reply:
As discussed, the is zero check on on the notional time weighted average position required config value. If the party's notional is zero, the event will still emit.